core/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -100,9 +133,52 @@ public final String path() {
 
     public abstract BytesReference content();
 
-    public abstract String header(String name);
+    /**
+     * Get the value of the header or {@code null} if not found. This method only retrieves the first header value if multiple values are
+     * sent. Use of {@link #getHeader(String)} should be preferred
+     */
+    public final String header(String name) {
+        List<String> values = headers.get(name);
```

Comment:
This is an implementation detail but currently the only real implementation in master is the Netty4 one and we use a wrapper around it to mimic a map. The key lookups are still case insensitive.